check_esxi_hardware.py

```diff
@@ -659,10 +659,17 @@ def handler(signum, frame):
   try:
     conntest = pywbem.WBEMConnection(hosturl, (user,password))
     c = conntest.EnumerateInstances('CIM_Card')
-  except:
+  except pywbem.cim_http.Error, arg:
     #raise
-    verboseoutput("Connection error, disable SSL certificate verification (probably patched pywbem)")
-    wbemclient = pywbem.WBEMConnection(hosturl, (user,password), no_verification=True)
+    if ( arg.__str__().find('SSL error: certificate verify failed') >= 0 ):
+      verboseoutput("Connection error, disable SSL certification verification (probably patched pywbem)")
+      wbemclient = pywbem.WBEMConnection(hosturl, (user,password), no_verification=True)
+    else:
+      verboseoutput("Other pywbem.cim_http.Error, continue as normal")
+      wbemclient = pywbem.WBEMConnection(hosturl, (user,password))
+  except:
```

Comment:
With v0.7.0, SSL errors are `pywbem.cim_operations.CIMError`s which are not subclasses of `pywbem.cim_http.Error` so they don't match the first `except`.

Reply:
Yep, agree. Not only in 0.7.0 but also later (tested here with 0.8.0-dev), but irrelevant in this if clause.